migrations/default/2_create_triggers/up.sql

```diff
@@ -0,0 +1,380 @@
+
+    -- for every key in the cris json object
+    for column_name in select jsonb_object_keys(new_cris_jb) loop
+        -- if the new value doesn't match the old
+        if(new_cris_jb -> column_name <> old_cris_jb -> column_name) then
+            -- see if the vz record has a value for this field
+            if (vz_record_jb ->> column_name is  null) then
+                -- this value is not overridden by VZ
+                -- so update the unified record with this new value
+                updates_todo := updates_todo || format('%I = $1.%I', column_name, column_name);
+            end if;
+        end if;
+    end loop;
+    if(array_length(updates_todo, 1) > 0) then
+        -- complete the update statement by joining all `set` clauses together
+        update_stmt := update_stmt
+            || array_to_string(updates_todo, ',')
+            || format(' where db.crashes.crash_id = %s', new.crash_id);
+        raise notice 'Updating unified crash record from CRIS update: %', update_stmt;
+        execute (update_stmt) using new;
+    else
+        raise notice 'No changes to unified record needed';
+    end if;
+    return null;
+end;
+$$;
+
+create trigger update_crash_from_cris_crash_update
+after update on db.crashes_cris for each row
+execute procedure db.crashes_cris_update();
+
+--
+-- handle a vz crash update by updating the
+-- unified crash record from cris + vz values
+--
+create or replace function db.crashes_vz_update()
+returns trigger
+language plpgsql
+as $$
+declare
+    new_vz_jb jsonb := to_jsonb (new);
+    old_vz_jb jsonb := to_jsonb (old);
+    cris_record_jb jsonb;
+    column_name text;
+    updates_todo text [] := '{}';
+    update_stmt text := 'update db.crashes set ';
+begin
+    -- get corresponding the cris record as jsonb
+    SELECT to_jsonb(crashes_cris) INTO cris_record_jb from db.crashes_cris where db.crashes_cris.crash_id = new.crash_id;
+    -- for every key in the vz json object
+    for column_name in select jsonb_object_keys(new_vz_jb) loop
+        --  create a set statement for the column
+        if cris_record_jb ? column_name then
+            -- if this column exists on the cris table, coalesce vz + cris values
+            updates_todo := updates_todo
+            || format('%I = coalesce($1.%I, cris_record.%I)', column_name, column_name, column_name);
+        else
+            updates_todo := updates_todo
+            || format('%I = $1.%I', column_name, column_name);
+        end if;
+    end loop;
+    -- join all `set` clauses together
+    update_stmt := update_stmt
+        || array_to_string(updates_todo, ',')
+        || format(' from (select * from db.crashes_cris where db.crashes_cris.crash_id = %s) as cris_record', new.crash_id)
+        || format(' where db.crashes.crash_id = %s ', new.crash_id);
+    raise notice 'Updating unified crash record from VZ update: %', update_stmt;
+    execute (update_stmt) using new;
+    return null;
+end;
+$$;
```

Comment:
worth nothing that the functions that fire on a `_vz` record update to not check to see if any values have changed. the working assumption is that if someone inserts into this table, we should accept that the unified record needs to be updated. 

that's in contrast to the `_cris` updates, where we definitely do want to block/ignore changes when the unified record will not change.

Reply:
I also don't think that it would be often that a VZ user would save the same value they previously chose without cancelling out, but it is possible to do in VZE where it fires a mutation when I just tested it.